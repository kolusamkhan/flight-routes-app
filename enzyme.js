const Adapter = require('enzyme-adapter-react-17-updated')
/**
 * Setup enzyme adapter class for testing env setup.
 */
require('enzyme').configure({adapter: new Adapter()})