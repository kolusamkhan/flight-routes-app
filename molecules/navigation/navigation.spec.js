import 'jsdom-global/register';
import * as React from 'react';
import expect from 'expect';
import { shallow, mount, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Navigation from './index';
import { NextIntlProvider, useTranslations } from 'next-intl';
global.__DEV__ = true;
jest.mock('next/router', () => ({
  useRouter: () => ({locale: 'en'})
}));
const message = require("../../locales/en.json");

describe('Navigation', () => {
  describe('Navigation component', () => {
    it('should render without throwing an error', function () {
      const wrapper = shallow(
        <NextIntlProvider messages={message}>
          <Navigation />
        </NextIntlProvider>);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    })
    
    it('should render Footer content', function () {
        const wrapper = mount(
          <NextIntlProvider messages={message}>
            <Navigation />
          </NextIntlProvider>);
        expect(wrapper.find('header')).toBeTruthy()
      })
  })  
})