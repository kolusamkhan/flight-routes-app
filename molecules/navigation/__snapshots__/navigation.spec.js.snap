// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Navigation component should render without throwing an error 1`] = `
<IntlProvider
  locale="en"
  messages={
    Object {
      "content": Object {
        "common": Object {
          "nav": Object {
            "home": "Home",
            "manageBooking": "Manage Booking",
            "myAccount": "My Account",
          },
          "navFooter": Object {},
          "title": "Fly Better",
        },
        "pageFlightRoutes": Object {
          "card": Object {
            "arrivalTitle": "Arrival",
            "ctaTitle": "Select",
            "departureTitle": "Departure",
          },
          "resultsTitle": "Search Results",
          "title": "Flight Routes",
        },
        "seo": Object {},
      },
    }
  }
>
  <Navigation />
</IntlProvider>
`;
